aptos/contracts/sources/governance/contract_upgrade.move

```diff
@@ -0,0 +1,69 @@
+module pyth::contract_upgrade {
+    use wormhole::cursor;
+    use pyth::deserialize;
+    use pyth::contract_upgrade_hash::{Self, Hash};
+    use pyth::state::{Self};
+    use std::vector;
+    use std::aptos_hash;
+    use aptos_framework::code;
+    use pyth::error;
+
+    const HASH_LENGTH: u64 = 32;
+
+    struct ContractUpgrade {
+        hash: Hash,
+    }
+
+    public fun execute(payload: vector<u8>) {
+        let ContractUpgrade {hash: hash} = from_byte_vec(payload);
+        state::set_contract_upgrade_authorized_hash(hash)
+    }
+
+    fun from_byte_vec(bytes: vector<u8>): ContractUpgrade {
+        let cursor = cursor::init(bytes);
+        let hash = contract_upgrade_hash::from_byte_vec(deserialize::deserialize_vector(&mut cursor, HASH_LENGTH));
+        cursor::destroy_empty(cursor);
+        
+        ContractUpgrade {
+            hash,
+        }
+    }
+
+    public entry fun do_contract_upgrade(
+        metadata_serialized: vector<u8>,
+        code: vector<vector<u8>>,
+    ) {
+        // Check to see if the given code matches the authorized hash
+        let hash = state::get_contract_upgrade_authorized_hash();
+        let c = copy code;
+        vector::reverse(&mut c);
+        let a = vector::empty<u8>();
+        while (!vector::is_empty(&c)) vector::append(&mut a, vector::pop_back(&mut c));
```

Comment:
seems like `a` is a copy of `c` before it was reversed? why is this code necessary? (maybe leave a code comment)